src/main/java/com/app/folioman/mfschemes/service/MfSchemeServiceImpl.java

```diff
@@ -145,113 +150,103 @@ private void processResponseEntity(Long schemeCode, NavResponse navResponse) {
-                    .filter(nav -> !mfFundScheme.getMfSchemeNavs().contains(nav))
-                    .toList();
-
-            LOGGER.info("No of entities to insert :{} for schemeCode/amfi :{}", newNavs.size(), schemeCode);
+            LOGGER.info("No of entries from API Server: {} for schemeCode/amfi: {}", newNavEntries.size(), schemeCode);
 
-            if (!newNavs.isEmpty()) {
+            // Use transaction to ensure database consistency
+            transactionTemplate.execute(status -> {
                 try {
-                    transactionTemplate.execute(status -> {
-                        // Fetch all existing NAVs for this scheme in a single query to avoid N+1 problem
-                        List<NavDateValueProjection> existingNavs =
-                                mfSchemeNavRepository.findAllNavDateValuesBySchemeId(mfFundScheme.getId());
-
-                        // Create a batch of NAVs that don't exist yet
-                        List<MFSchemeNav> navsToSave = newNavs.stream()
-                                .filter(newNav -> {
-                                    // Create a projection for comparison
-                                    NavDateValueProjection projection =
-                                            new NavDateValueProjection(newNav.getNav(), newNav.getNavDate());
-                                    return !existingNavs.contains(projection);
-                                })
-                                .peek(newNav -> newNav.setMfScheme(mfFundScheme))
-                                .toList();
-
-                        if (!navsToSave.isEmpty()) {
-                            LOGGER.info("Saving {} NAVs in batch for scheme {}", navsToSave.size(), schemeCode);
-                            try {
-                                // Save all NAVs in a single batch operation
-                                mfSchemeNavRepository.saveAll(navsToSave);
-                            } catch (DataIntegrityViolationException ex) {
-                                // When batch insert fails, use a partitioning approach instead of individual saves
-                                LOGGER.warn("Batch insert failed due to constraint violations: {}", ex.getMessage());
-
-                                // Use a binary partitioning approach to save NAVs in smaller batches
-                                processNavsInPartitions(navsToSave, mfFundScheme.getId());
-                            }
-                        } else {
-                            LOGGER.info(
-                                    "All {} NAVs already exist in database for scheme {}", newNavs.size(), schemeCode);
-                        }
-
+                    // Fetch all existing NAV dates for this scheme in a single query
+                    // This avoids loading full entity objects when we only need dates for comparison
+                    List<NavDateValueProjection> existingNavs =
```

Comment:
@coderabbitai, can I move this fetch outside of the current transaction which is not suited for readonly transaction